pkg/multiboot/ranges_linux.go

```diff
@@ -0,0 +1,276 @@
+// Copyright 2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package multiboot
+
+import (
+	"bytes"
+	"debug/elf"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/u-root/u-root/pkg/kexec"
+)
+
+var ErrNotEnoughSpace = errors.New("not enough space")
+
+var sizeofMmap = uint32(binary.Size(Mmap{}))
+
+type segments []kexec.Segment
+
+type rangeType string
+
+const (
+	rangeRAM     rangeType = "System RAM"
+	rangeDefault           = "Default"
+	rangeNVACPI            = "ACPI Non-volatile Storage"
+	rangeACPI              = "ACPI Tables"
+	rangeNVS               = "Reserved"
+)
+
+var rangeTypes = map[rangeType]uint32{
+	rangeRAM:     1,
+	rangeDefault: 2,
+	rangeNVACPI:  3,
+	rangeACPI:    3,
+	rangeNVS:     4,
+}
+
+type memoryRange struct {
+	start, end uintptr
+	typ        rangeType
+}
+
+type memoryRanges []memoryRange
+
+// Mmap represents single mmap entry.
```

Comment:
can we write this out as MemoryMap? i feel mmap is easily confused with the linux mmap system call.

And I assume this corresponds to the multiboot mmap structure, does it? I'd expect something like

```
// MemoryMap represents a reserved range of memory passed via the Multiboot Info header.
```